### PR TITLE
[transactional tests] Fixed missing input object checks

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

Input checker was not run before execution in transactional tests. The programmable transaction runtime relies on these checks

## Test Plan 

Just test fixes 

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-core/src/transaction_input_checker.rs [hunk 1]
```diff
@@ -217,7 +217,7 @@ async fn check_gas(
 /// Check all the objects used in the transaction against the database, and ensure
 /// that they are all the correct version and number.
 #[instrument(level = "trace", skip_all)]
-async fn check_objects(
+pub fn check_objects(
```
Comment (tnowacki) [inline crates/sui-core/src/transaction_input_checker.rs:220]: @mystenmark, tagging you incase you have concerns of this being public

**Review by dariorussi — APPROVED** [review]
sounds very reasonable to me and even more correct.
Am I missing something?

File: crates/sui-core/src/transaction_input_checker.rs (hunk 1 above)
Comment (mystenmark) [reply crates/sui-core/src/transaction_input_checker.rs:220]: I don't see why this would be a problem

**Comment by tnowacki** [comment]
> sounds very reasonable to me and even more correct. Am I missing something?

The thing missing is that it is still a bit messy that I don't actually call the transaction input checker `check_transaction_input` (I think I have the name right), and instead I have re-implemented parts of that function for tests. 
Which sucks, but is fine for now.

**Pull request merged**